### PR TITLE
Nest tests under a span

### PR DESCRIPTION
**Pull request by matt-richardson**
# Background

It's kinda annoying/odd that tests are directly under a build span.

# Results

This nests them under a `Tests` span, if there are any tests.

We also no longer duplicate tests:
* we were emiting them as part of the build log parsing
* we were also emitting them from composite builds

## After

image

# How to review this PR

# Pre-requisites
- [ ] I have considered informing or consulting the right people
- [ ] I have considered appropriate testing for my change.

### CODE REVIEW

File: server/src/main/resources/buildServerResources/projectConfigurationSettingsHoneycomb.jspf
Comment (cailyoung) [inline server/src/main/resources/buildServerResources/projectConfigurationSettingsHoneycomb.jspf:1]: Is this meant to be here?

Comment (matt-richardson) [reply server/src/main/resources/buildServerResources/projectConfigurationSettingsHoneycomb.jspf:1]: No, no it's not...
🤦

File: server/src/main/java/com/octopus/teamcity/opentelemetry/server/TeamCityBuildListener.java [hunk 1]
```diff
@@ -308,9 +320,13 @@ private void createBuildStepSpans(SRunningBuild build, Span buildSpan) {
     }
 
     private void createBlockMessageSpan(BlockLogMessage blockLogMessage, Span buildSpan, Map<String, Span> blockMessageSpanMap, SRunningBuild build) {
+        if (blockLogMessage.getBlockType().equals("$TEST_BLOCK$")) {
+            //we handle these explicitly when we publish tests (in createTestExecutionSpans)
```
Comment (cailyoung) [inline server/src/main/java/com/octopus/teamcity/opentelemetry/server/TeamCityBuildListener.java:324]: Didn't we have an external user relying on these?

File: server/src/main/java/com/octopus/teamcity/opentelemetry/server/TeamCityBuildListener.java [hunk 2]
```diff
@@ -298,8 +310,8 @@ private void createQueuedEventsSpans(SRunningBuild build, Span buildSpan) {
     }
 
     private void createBuildStepSpans(SRunningBuild build, Span buildSpan) {
+        if (build.isCompositeBuild()) return;
```
Comment (cailyoung) [inline server/src/main/java/com/octopus/teamcity/opentelemetry/server/TeamCityBuildListener.java:313]: Sensible.

File: server/src/main/java/com/octopus/teamcity/opentelemetry/server/TeamCityBuildListener.java [hunk 3]
```diff
@@ -57,16 +56,12 @@ public TeamCityBuildListener(
 
     @Override
     public void buildStarted(@NotNull SRunningBuild build) {
-        LOG.debug(String.format("Build started method triggered for %s, id %d", getBuildName(build), build.getBuildId()));
-
-        if (!nodesService.getCurrentNode().isMainNode())
-        {
-            LOG.debug(String.format("This is not the main node - skipping OTEL tracing for %s (id %d) and leaving that responsibility for the main node to deal with", getBuildName(build), build.getBuildId()));
```
Comment (cailyoung) [inline server/src/main/java/com/octopus/teamcity/opentelemetry/server/TeamCityBuildListener.java:64]: Thanks. Watch the log ingest costs go down!

**Review by cailyoung — APPROVED** [review]
Nice work. One non-blocking question (we may have to semver bump minor)

File: server/src/main/java/com/octopus/teamcity/opentelemetry/server/TeamCityBuildListener.java (hunk 1 above)
Comment (matt-richardson) [reply server/src/main/java/com/octopus/teamcity/opentelemetry/server/TeamCityBuildListener.java:324]: Are you thinking of https://github.com/OctopusDeploy/opentelemetry-teamcity-plugin/pull/147?

**Comment by matt-richardson** [comment]
[sc-87109]

**Pull request merged**